.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,62 @@
+name: Build and publish
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      docker_tag_prefix:
+        description: "Enter the Docker Tag Prefix (e.g adhoc-test)"
+        required: true
+        type: string
+
+env:
+  ECR_REPOSITORY_URL: 673156464838.dkr.ecr.us-west-2.amazonaws.com
+  ECR_REPOSITORY_NAME: gpt-survey-summarizer
+  DOCKER_TAG_PREFIX: ${{ github.event.inputs.docker_tag_prefix }}
+
+jobs:
+  build-and-publish:
+    name: Build and Publish Docker Image
+    runs-on: minafoundation-default-runners
+    steps:
+
+      - name: 📥 Checkout
+        uses: actions/checkout@v4
+
+      - name: 🦀 Get application version from package.json
+        id: application-version
+        uses: martinbeentjes/npm-get-version-action@v1.3.1
+
+      - name: 🏷️ Generate Tag
+        run: |
+          SHORT_SHA=$(echo $GITHUB_SHA | cut -c1-7)
+          if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+            echo "TAG=${{ env.DOCKER_TAG_PREFIX }}-${{ steps.application-version.outputs.current-version }}-$SHORT_SHA" >> $GITHUB_ENV
+          elif [ "${{ github.event_name }}" == "push" ] && [ -n "${{ github.event.ref }}" ]; then
+            echo "TAG=${{ steps.application-version.outputs.current-version }}-$SHORT_SHA" >> $GITHUB_ENV
+          else
+            echo "Invalid event. Exiting..."
+            exit 1
+          fi
+
+      - name: 🔑 ECR Login
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
```

Comment:
**suggestion:** I just noticed that the actions are raising warnings [here](https://github.com/MinaFoundation/gptSurveySummarizer/actions/runs/8815662421). We should be able to fix this by upgrading the versions of the `amazon-ecr-login` action.